### PR TITLE
Add terraform transition example

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

Example of initial Composition that contains only tf workspace with eventual transition to a mixes scenario and potential full migration to Crossplane-native resources

Signed-off-by: Yury Tsarev <yury@upbound.io>
I have:

- [ ] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

Buy applying Composition step by step

### CODE REVIEW

File: examples/transition/xsubnet.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: aws.platformref.upbound.io/v1alpha1
+kind: XSubnet
+metadata:
+  name: subnet-with-vpc
```
Comment (luebken) [inline examples/transition/xsubnet.yaml:4]: I tend to use `my-subnet` and `my-vpc` for these type of example where I'm refering to something the user has to change.

**Review by luebken — APPROVED** [review]
LGTM

File: examples/transition/xsubnet.yaml (hunk 1 above)
Comment (ytsarev) [reply examples/transition/xsubnet.yaml:4]: Thanks for the feedback, it makes perfect sense for the step-by-step tutorials. Meanwhile, in standalone examples (this one might be used outside of blogpost), I prefer to keep the naming as neutral as possible.

**Review by bobh66 — APPROVED** [review]

File: examples/transition/00-mr-tf-workspace/workspace-inline.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+apiVersion: tf.upbound.io/v1beta1
+kind: Workspace
+metadata:
+  name: sample-inline
+spec:
+  forProvider:
+    source: Inline
+    module: |
+      resource "aws_vpc" "main" {
```
[the remark this answers is not in the document]
Comment (ytsarev) [reply examples/transition/00-mr-tf-workspace/workspace-inline.yaml:9]: Yes, but I rely on the usual practice of region being handled in ProviderConfig of provider-terraform like in another example of https://github.com/upbound/provider-terraform/blob/main/examples/providerconfig-aws.yaml#L22-L25

**Pull request merged**